pallets/gear/src/tests.rs

```diff
@@ -5404,6 +5408,176 @@ fn uninitialized_program_terminates_on_pause() {
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_2),
+            pay_rent_id,
+            Kind::PayProgramRent(pay_rent_id.into_origin().into(), rent).encode(),
+            20_000_000_000,
+            0,
+        ));
+
+        run_to_next_block(None);
+
+        let program = ProgramStorageOf::<Test>::get_program(pay_rent_id)
+            .and_then(|p| ActiveProgram::try_from(p).ok())
+            .expect("program should exist");
+        let expiration_block = program.expiration_block;
+        assert_eq!(old_block + u64::from(block_count), expiration_block);
+
+        // attempt to pay rent for not existing program
+        let pay_rent_account_id = AccountId::from_origin(pay_rent_id.into_origin());
+        let balance_before = Balances::free_balance(pay_rent_account_id);
+
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_2),
+            pay_rent_id,
+            Kind::PayProgramRent([0u8; 32], rent).encode(),
+            20_000_000_000,
+            0,
+        ));
+
+        run_to_next_block(None);
+
+        assert_eq!(balance_before, Balances::free_balance(pay_rent_account_id));
+
+        // try to pay greater rent than available value
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_2),
+            pay_rent_id,
+            Kind::PayProgramRent(pay_rent_id.into_origin().into(), program_value).encode(),
+            20_000_000_000,
+            0,
+        ));
+
+        run_to_next_block(None);
+
+        assert_eq!(balance_before, Balances::free_balance(pay_rent_account_id));
+        let program = ProgramStorageOf::<Test>::get_program(pay_rent_id)
+            .and_then(|p| ActiveProgram::try_from(p).ok())
+            .expect("program should exist");
+        assert_eq!(expiration_block, program.expiration_block);
+
+        // pay maximum possible rent
+        let block_count: BlockNumberFor<Test> = u32::MAX.into();
+        let rent = Gear::rent_fee_for(block_count);
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_2),
+            pay_rent_id,
+            Kind::PayProgramRent(pay_rent_id.into_origin().into(), rent).encode(),
+            20_000_000_000,
+            rent,
+        ));
+
+        run_to_next_block(None);
+
+        assert_eq!(balance_before, Balances::free_balance(pay_rent_account_id));
+        let program = ProgramStorageOf::<Test>::get_program(pay_rent_id)
+            .and_then(|p| ActiveProgram::try_from(p).ok())
+            .expect("program should exist");
+        assert_eq!(
+            expiration_block.saturating_add(block_count),
+            program.expiration_block
+        );
+        assert!(TaskPoolOf::<Test>::contains(
+            &program.expiration_block,
+            &ScheduledTask::PauseProgram(pay_rent_id)
+        ));
+    });
+}
+
+#[test]
+fn pay_program_rent_extrinsic_works() {
```

Comment:
case of pay more for u32::MAX (refund) case?

Reply:
frankly speaking this is a bit tricky since `BlockNumber = u64` for Test runtime )